libtiledbsoma/src/soma/managed_query.h

```diff
@@ -1086,6 +1104,15 @@ bool ManagedQuery::_extend_and_evolve_schema_and_write<std::string>(
     Enumeration enmr,
     ArraySchemaEvolution& se);
 
+template <>
+bool ManagedQuery::_extend_and_evolve_schema<std::string>(
```

Comment:
nit: not clear to me why this templating is necessary at this level -- I _think_ the only true specialization is in `remap_indexes`, but might have missed something.

Reply:
@ihnorton this was carefully and thoroughly discussed here as a team effort:
https://github.com/single-cell-data/TileDB-SOMA/pull/3888

Reply:
What we could perhaps do is to move the `switch` on TileDB type from here
https://github.com/single-cell-data/TileDB-SOMA/blob/2609d8b4545a1d296def5bdc7a4b496429933221/libtiledbsoma/src/soma/managed_query.cc#L1125-L1166
down into a non-templated `ManagedQuery::_extend_and_evolve_schema`

Reply:
This is fine to discuss in follow-up, the only impact is code gen time / binary size.

Reply:
No objection, we can have a separate discussion about where to draw the line between generic and type-specialized code.